### PR TITLE
Ros model parser

### PR DESCRIPTION
**Pull request by hsd-dev**
Removes dependency on `ros_graph_parser` and uses `ros_model_parser` instead.  
Improved functions to extract interfaces from the ROS graph.  

Depends on https://github.com/ipa320/ros_model_parser/pull/11

### CODE REVIEW

File: .rosinstall [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- git:
+    local-name: ros_model_parser
+    uri: https://github.com/ipa320/ros_model_parser.git
+    version: master
+- git:
+    local-name: rosgraph_monitor
+    uri: https://github.com/ipa-hsd/rosgraph_monitor.git
```
Comment (NadiaHG) [inline .rosinstall]: why this?

**Review by ipa-nhg — COMMENTED** [review]
could you please add a short documentation about how to test these changes? Specially the test of the rosgraph introspection

File: .rosinstall (hunk 1 above)
Comment (hsd-dev) [reply .rosinstall]: It is this one: https://github.com/ipa320/rosgraph_monitor/pull/1. Will update this once the PR is merged.

**Comment by hsd-dev** [comment]
### Setup the workspace

```
mkdir ~/model_ws && cd ~/model_ws
wstool init src https://raw.githubusercontent.com/ipa-hsd/rosgraph_monitor/ros_model_parser/.rosinstall
source /opt/ros/$ROS_DISTRO/setup.bash
rosdep update && rosdep install --from-paths ./src -y -i -r
catkin build
```

To generate rossystem model from the ROS graph, launch a system of choice:
```
source ~/model_ws/devel/setup.bash
export TURTLEBOT3_MODEL=waffle
roslaunch turtlebot3_gazebo turtlebot3_empty_world.launch
```
Start the `rosgraph_monitor` and listen to `/diagnostics` topic  
(`source ~/model_ws/devel/setup.bash` on each terminal)
```
# Terminal 1:
rosrun rosgraph_monitor monitor

# Terminal 2:
rosservice call /load_observer "name: 'ROSGraphObserver'"

#Terminal 3:
rostopic echo /diagnostics
```

**Review by ipa-nhg — APPROVED** [review]
Thanks for the instructions, I tested the commands together with the https://github.com/ipa320/ros_model_parser/pull/14
 version of ros_model_parser and works as expected 👍🏽 
 
 Please remove the lines https://github.com/ipa320/rosgraph_monitor/blob/ecf781b073c179d7ffaa455fd94dc8393b9747e4/.rosinstall#L5-L8 
 
 And modify https://github.com/ipa-hsd/rosgraph_monitor/blob/ros_model_parser/src/rosgraph_monitor/observers/graph_observer.py#L172 to get the path as input and not hard-coded 
 
 Thanks a lot this is a big step forward!  🎉

**Comment by hsd-dev** [comment]
> Please remove the lines

I can modify it to `ipa320/master`. In that case no need to clone `rosgraph_monitor` separately

**Comment by hsd-dev** [comment]
> And modify https://github.com/ipa-hsd/rosgraph_monitor/blob/ros_model_parser/src/rosgraph_monitor/observers/graph_observer.py#L172 to get the path as input and not hard-coded

See: https://github.com/ipa320/rosgraph_monitor/pull/1/commits/01e3d6aa8b7b50809d90744cc99f20065dba4810
 @ipa-nhg is it fine that it expects a fully resolved path?

**Comment by hsd-dev** [comment]
Merging this since most of the issues are addressed. Issues will be addressed in new PR.

**Pull request merged**